File: Lecture03/ButtonsAndText/app/src/main/java/com/codingblocks/buttonsandtext/MainActivity.java
package com.codingblocks.buttonsandtext;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText etVar1, etVar2;
    Button btnAdd;
    TextView tvResult;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etVar1 = findViewById(R.id.etVar1);
        etVar2 = findViewById(R.id.etVar2);
        btnAdd = findViewById(R.id.btnAdd);
        tvResult = findViewById(R.id.tvResult);

        btnAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int var1 = Integer.valueOf(etVar1.getText().toString());
                int var2 = Integer.valueOf(etVar2.getText().toString());
                int result = var1 + var2;

                tvResult.setText(String.valueOf(result));
            }
        });


    }
}

File: Lecture18/UnitTesting/app/src/main/java/com/codingblocks/unittesting/MainActivity.kt
package com.codingblocks.unittesting

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnCalcFare.setOnClickListener {
            val km = etKm.text.toString().toFloat()
            val min = etMin.text.toString().toInt()

            val fare = calcFare(km, min)
            tvResult.text = fare.toString()
        }
        if (intent.getStringExtra("RESULT") != null) {
            tvResult.text = intent.getStringExtra("RESULT")
        }
    }

    companion object {
        @JvmStatic
        fun calcFare(km: Float, min: Int): Float {
            var fare = 60f
            if (km > 5) fare += (km -5) * 10
            if (min > 30) fare += (min - 30) * 2
            return fare
        }
    }
}
